commit 6fe2d789e3a58306cf38e3a426eed52999e7831d
Author: Benjamin Vedder <benjamin@vedder.se>
Date:   Mon Oct 20 19:44:49 2025 +0200

    Persist CAN ID and CAN Baud Rate across firmware updates

diff --git a/conf_general.c b/conf_general.c
index 7d4a0106..fbd4b9d1 100644
--- a/conf_general.c
+++ b/conf_general.c
@@ -127,11 +127,26 @@ __attribute__((section(".text2"))) void conf_general_init(void) {
 		if (g_backup.hw_config_init_flag == BACKUP_VAR_INIT_CODE) {
 			memcpy((void*)backup_tmp.hw_config, (uint8_t*)g_backup.hw_config, sizeof(g_backup.hw_config));
 		}
+
+		if (g_backup.enc_corr_init_flag == BACKUP_VAR_INIT_CODE) {
+			memcpy((void*)backup_tmp.enc_corr, (uint8_t*)g_backup.enc_corr, sizeof(g_backup.enc_corr));
+			backup_tmp.enc_corr_en = g_backup.enc_corr_en;
+		}
+
+		if (g_backup.can_init_flag == BACKUP_VAR_INIT_CODE) {
+			backup_tmp.can_baud = g_backup.can_baud;
+			backup_tmp.can_id = g_backup.can_id;
+		} else {
+			backup_tmp.can_baud = APPCONF_CAN_BAUD_RATE;
+			backup_tmp.can_id = HW_DEFAULT_ID;
+		}
 	}
 
 	backup_tmp.odometer_init_flag = BACKUP_VAR_INIT_CODE;
 	backup_tmp.runtime_init_flag = BACKUP_VAR_INIT_CODE;
 	backup_tmp.hw_config_init_flag = BACKUP_VAR_INIT_CODE;
+	backup_tmp.enc_corr_init_flag = BACKUP_VAR_INIT_CODE;
+	backup_tmp.can_init_flag = BACKUP_VAR_INIT_CODE;
 
 	g_backup = backup_tmp;
 	conf_general_store_backup_data();
@@ -347,6 +362,8 @@ __attribute__((section(".text2"))) void conf_general_read_app_configuration(app_
 	// Set the default configuration
 	if (!is_ok) {
 		confgenerator_set_defaults_appconf(conf);
+		conf->can_baud_rate = g_backup.can_baud;
+		conf->controller_id = g_backup.can_id;
 	}
 }
 
@@ -403,6 +420,10 @@ __attribute__((section(".text2"))) bool conf_general_store_app_configuration(app
 	mc_interface_ignore_input_both(100);
 	utils_sys_unlock_cnt();
 
+	g_backup.can_id = conf->controller_id;
+	g_backup.can_baud = conf->can_baud_rate;
+	conf_general_store_backup_data();
+
 	return is_ok;
 }
 
